domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/frequencies/Frequency.java

```diff
@@ -0,0 +1,218 @@
+         * @param headwaySecs identifies times in seconds between departures in the same headway for a trip service
+         * @return builder for future object creation
+         */
+        public FrequencyBuilder headwaySecs(@NotNull Integer headwaySecs) {
+            this.headwaySecs = headwaySecs;
+            return this;
+        }
+
+        /**
+         * Sets field exactTimes value and returns this
+         *
+         * @param exactTimes identifies an end time for a trip
+         * @return builder for future object creation
+         */
+        public FrequencyBuilder exactTimes(@Nullable Integer exactTimes) {
+            this.exactTimes = ExactTimes.fromInt(exactTimes);
+            this.originalExactTimes = exactTimes;
+            return this;
+        }
+
+        public EntityBuildResult<?> build() {
+            noticeCollection.clear();
+            if (tripId == null || startTime == null || endTime == null || headwaySecs == null ||
+                    headwaySecs < 0 || exactTimes == null) {
+
+                if (tripId == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("frequencies.txt",
+                            "trip_id", tripId));
+                }
+
+                if (startTime == null && originalStartTime == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("frequencies.txt",
+                            "start_time", tripId));
+                }
+                //else if (startTime == null && originalStartTime != null) InvalidTimeNotice?
+
+                if (endTime == null && originalEndTime == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("frequencies.txt",
+                            "end_time", tripId));
+                }
+                //else if (endTime == null && originalEndTime != null) InvalidTimeNotice?
+
+                if (headwaySecs == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("frequencies.txt",
+                            "headway_secs", tripId));
+                } else if (headwaySecs < 0) {
+                    noticeCollection.add(new IntegerFieldValueOutOfRangeNotice("frequencies.txt",
+                            "headway_secs", tripId, 0, Integer.MAX_VALUE, headwaySecs));
+                }
+
+                // the following statement is true when ExactTimes.isEnumValueValid(originalTransferInteger)
+                // returns false
+                if (exactTimes == null) {
+                    noticeCollection.add(new UnexpectedEnumValueNotice("frequencies.txt",
+                            "exact_times", tripId, originalExactTimes));
+                }
+                return new EntityBuildResult<>(noticeCollection);
+            }
+            return new EntityBuildResult<>(new Frequency(tripId, startTime, endTime, headwaySecs, exactTimes));
+        }
+    }
+
+    public static Integer convertTimeToInteger(String timeString) {
```

Comment:
This needs a lot of unit tests :)

Reply:
This method also probably belongs in a utility class somewhere, as you'll need it for stop_times.txt as well.

Reply:
I agree. For now, we discussed that it should be move into the GtfsEntityParser, so the internal representation of an entity with a time type would implement the Integer form. I moved the code as is. I suggest a TimeType class or similar.